pygmt/datasets/samples.py

```diff
@@ -290,3 +217,140 @@ def _load_earth_relief_holes():
     """
     fname = which("@earth_relief_20m_holes.grd", download="c")
     return load_dataarray(fname, engine="netcdf4")
+
+
+class GMTSampleData(NamedTuple):
+    """
+    Information of a sample dataset.
+
+    Attributes
+    ----------
+    func : callable
+        The function that loads the sample dataset.
+    description : str
+        The description of the sample dataset.
+    """
+
+    func: Callable
+    description: str
+
+
+datasets = {
+    "bathymetry": GMTSampleData(
+        func=_load_baja_california_bathymetry,
+        description="Table of ship bathymetric observations off Baja California",
+    ),
+    "earth_relief_holes": GMTSampleData(
+        func=_load_earth_relief_holes,
+        description="Regional 20 arc-minutes Earth relief grid with holes",
+    ),
+    "fractures": GMTSampleData(
+        func=_load_fractures_compilation,
+        description="Table of hypothetical fracture lengths and azimuths",
+    ),
+    "hotspots": GMTSampleData(
+        func=_load_hotspots,
+        description="Table of locations, names, and symbol sizes of hotpots from "
+        "Mueller et al., 1993",
+    ),
+    "japan_quakes": GMTSampleData(
+        func=_load_japan_quakes,
+        description="Table of earthquakes around Japan from NOAA NGDC database",
+    ),
+    "mars_shape": GMTSampleData(
+        func=_load_mars_shape,
+        description="Table of topographic signature of the hemispheric dichotomy of "
+        "Mars from Smith and Zuber (1996)",
+    ),
+    "maunaloa_co2": GMTSampleData(
+        func=_load_maunaloa_co2,
+        description="Table of CO2 readings from Mauna Loa",
+    ),
+    "notre_dame_topography": GMTSampleData(
+        func=_load_notre_dame_topography,
+        description="Table 5.11 in Davis: Statistics and Data Analysis in Geology",
+    ),
+    "ocean_ridge_points": GMTSampleData(
+        func=_load_ocean_ridge_points,
+        description="Table of ocean ridge points for the entire world",
+    ),
+    "rock_compositions": GMTSampleData(
+        func=_load_rock_sample_compositions,
+        description="Table of rock sample compositions",
+    ),
+    "usgs_quakes": GMTSampleData(
+        func=_load_usgs_quakes,
+        description="Table of earthquakes from the USGS",
+    ),
+}
+
+
+def fmt_dataset_list(module_func):
+    """
+    Decorator to replace the ``{dataset_list}`` placeholder in the docstring.
+
+    Parameters
+    ----------
+    module_func : function
+        The function to decorate.
+
+    Returns
+    -------
+    module_func
+        The same *module_func* but with the docstring formatted.
+    """
+    texts = "\n        ".join(
```

Comment:
```suggestion
    """
    # Add a new lines and 8 white spaces for formatting in the documentation
    texts = "\n        ".join(
```

I don't know of a better way to fix this problem for documentation displays, and I think using `\n        ` works since this runs behind the scenes.  I do think there should be a comment explaining why there is a strange join string.